### ISSUE TITLE
Ports snap weird with connected rows of ports

### ISSUE THREAD

**Issue by LeonMontealegre** [OpenCircuits/OpenCircuits#466]
To reproduce:

Create an IC with ports slightly farther-apart then default for an IC

then bus the ports with an Encoder and try to snap the bottom two together.

This happens:

![bug2](https://user-images.githubusercontent.com/29997057/70365026-4b259200-185d-11ea-8fed-8c20a254aeaa.gif)

**Comment by Robert-Hammond** [comment]
I don't see how this is a bug, it's due to the distance between ports being wider on one component than the other.

**Comment by LeonMontealegre** [comment]
@Robert-Hammond yes, but intuitively, the bottom rows are going to be aligned first and therefore should be the ones that snap. But the current behavior is that it causes some sort of chain-of-snaps and then the top row is aligned.
